### PR TITLE
Fix pika's Record Mixing PR

### PR DESCRIPTION
**Pull request by DizzyEggg**
This file is mad, will maybe attempt at decompiling more of it once the structs and variables are better known.

#107 can be closed after this one gets merged.

### CODE REVIEW

File: include/mystery_event_script.h [hunk 1]
```diff
@@ -6,5 +6,6 @@ bool32 sub_8153884(u32 *a0);
 u32 RunMysteryEventScript(u8 *script);
 void SetMysteryEventScriptStatus(u32 val);
 u16 GetRecordMixingGift(void);
+u16 GetRecordMixingGift(void);
```
Comment (Diegoisawesome) [inline include/mystery_event_script.h]: Extra declaration

File: constants/item_data_constants.inc [hunk 2]
```diff
@@ -0,0 +1,5 @@
+	.set POCKET_ITEMS,      0x01
```
Comment (Diegoisawesome) [inline constants/item_data_constants.inc]: Figure out if this file can be deleted

File: src/record_mixing.c [hunk 3]
```diff
@@ -1,28 +1,1582 @@
...
+    gUnknown_03001148 = &gUnknown_02039F9C;
+    gUnknown_0300114C = gSaveBlock2Ptr->field_64C;
+    gUnknown_03001150 = &gSaveBlock1Ptr->lilycoveLady;
+    gUnknown_03001154 = gSaveBlock2Ptr->field_0DC;
+    gUnknown_03001158 = gSaveBlock2Ptr->field_64C;
+}
+
+void sub_80E6CA0(struct PlayerRecords *dest)
+{
+    memcpy(dest->secretBases, gUnknown_03001134, sizeof(struct SecretBaseRecord) * 20);
```
Comment (Diegoisawesome) [inline src/record_mixing.c]: These pointers should probably be made into arrays of the length that's being multiplied here, then use sizeof(*ptr)

File: src/record_mixing.c [hunk 4]
```diff
@@ -1,28 +1,1582 @@
...
+            sub_80E7AA4(src, recordSize, sp24, gUnknown_0858CFBE[r1_80e7b54][2], gUnknown_0858CFBE[r1_80e7b54][3]);
+            break;
+    }
+    _src = (void *)src + which * recordSize;
+    memcpy(&gSaveBlock1Ptr->daycare.mons[0].misc.mail, &_src->unk_00[0], sizeof(struct DayCareMailRecordMixing));
+    memcpy(&gSaveBlock1Ptr->daycare.mons[1].misc.mail, &_src->unk_00[1], sizeof(struct DayCareMailRecordMixing));
+    SeedRng(oldSeed);
+}
+#else
+__attribute__((naked)) static void sub_80E7B60(struct UnkStruct_80E7B60 *src, size_t recordSize, u8 which, TVShow *shows)
```
Comment (Diegoisawesome) [inline src/record_mixing.c]: Use NAKED

File: src/record_mixing.c [hunk 5]
```diff
@@ -1570,13 +1574,72 @@ static void sub_80E8110(struct UnkRecordMixingStruct *arg0, struct UnkRecordMixi
...
+void sub_80E8260(struct UnkRecordMixingStruct2 *dst)
+{
+    s32 i, j;
+
+    for (i = 0; i < 9; i++)
+    {
+        for (j = 0; j < 2; j++)
+        {
+            CopyUnalignedWord(dst->field_0[i][j].playerId, gSaveBlock2Ptr->playerTrainerId);
+            dst->field_0[i][j].language = LANGUAGE_ENGLISH;
```
Comment (Diegoisawesome) [inline src/record_mixing.c]: Should this be replaced with the GAME_LANGUAGE define? For future portability?

**Comment by m-flak** [comment]
```
Portability 
```
You are being rhetorical,  yes?

**Comment by Diegoisawesome** [comment]
Lol well I mean to make it easy to change the game's language with one define later, for hacks.

**Pull request merged**